client/extensions/woocommerce/index.js

```diff
@@ -151,15 +155,23 @@ function addStorePage( storePage, storeNavigation ) {
 	} );
 }
 
-function createStoreNavigation( context, next ) {
-	const state = context.store.getState();
-	const selectedSite = getSelectedSite( state );
+function mapStateToProps( state ) {
+	let siteId = getSelectedSiteId( state );
+	if ( ! siteId && ( 1 === getCurrentUserSiteCount( state ) ) ) {
+		siteId = getPrimarySiteId( state );
+	}
+	const site = getSite( state, siteId );
 
+	return {
+		site
+	};
+}
+
+function createStoreNavigation( context, next ) {
 	renderWithReduxStore(
 		React.createElement(
-			StoreSidebar, {
+			connect( mapStateToProps )( StoreSidebar ), {
```

Comment:
Um I did this because I thought you had a good suggestion here:

 https://github.com/Automattic/wp-calypso/pull/14022#discussion_r116342627

In order to pass the site into the sidebar as a prop, and have that prop come from state (and thereby keep the sidebar "in sync", is this not the only way?